src/main/java/baritone/process/MineProcess.java

```diff
@@ -102,10 +102,17 @@ public PathingCommand onTick(boolean calcFailed, boolean isSafeToCancel) {
                 return null;
             }
         }
+
         if (!Baritone.settings().allowBreak.value) {
-            logDirect("Unable to mine when allowBreak is false!");
-            cancel();
-            return null;
+            this.filter = new BlockOptionalMetaLookup(filter.blocks()
+                .stream()
+                .filter(e -> Baritone.settings().allowBreakAnyway.value.contains(e.getBlock()))
+                .toArray(BlockOptionalMeta[]::new));
+            if (this.filter.blocks().isEmpty()) {
+                logDirect("Unable to mine when allowBreak is false!");
+                cancel();
+                return null;
+            }
```

Comment:
I don't think pruning every tick is a good idea because it causes inconsistent behavior when `allowBreakAnyway` is changed. Either we prune once at the start and ignore changes to `allowBreakAnyway` or we filter the original list every tick so both removals from and additions to `allowBreakAnyway` are picked up.
Since `MineProcess` also immediately responds to other setting changes I think filtering is the way to go. Not sure whether it is worth the effort to cache filtered list (and a copy of the setting value) to avoid refiltering.

Reply:
did you make some kind of setting change watcher in one of your pr's? but yeah, I guess I should store the original for refiltering each tick at least

Reply:
Regardless of whether we attempt to cache it (no, I have not yet implemented a setting watcher), the filtering should happen in `rescan`, not in `onTick`.